pgx/src/datum/from.rs

```diff
@@ -10,36 +10,56 @@ Use of this source code is governed by the MIT license that can be found in the
 //! for converting a pg_sys::Datum and a corresponding "is_null" bool into a typed Option
 
 use crate::{
-    pg_sys, text_to_rust_str_unchecked, varlena_to_byte_slice, AllocatedByPostgres, PgBox,
-    PgMemoryContexts,
+    pg_sys, text_to_rust_str_unchecked, varlena_to_byte_slice, AllocatedByPostgres, IntoDatum,
+    PgBox, PgMemoryContexts,
 };
 use std::ffi::CStr;
+use std::num::NonZeroUsize;
 
-/// Convert a `(pg_sys::Datum, is_null:bool, type_oid:pg_sys::Oid)` tuple into a Rust type
+/// If converting a Datum to a Rust type fails, this is the set of possible reasons why.
+#[derive(Debug)]
```

Comment:
Anything returned in a `Result<T, E>` as the `E` type should always `impl Error`.  I suggest using `thiserror` which interacts well with `eyre` and `anyhow` to `#[derive(Error)]` this and similar types: https://github.com/dtolnay/thiserror

Reply:
Done.  Again, just `impl Error` -- no importing of more dependencies.